### PR TITLE
My db

### PR DESCRIPTION
**Pull request by astroboyReloaded**
### Project requirements
- Add a column species of type string to your animals table. Modify your `schema.sql` file.
- Insert the following data:
   - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
   - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
   - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
   - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
   - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
   - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
   - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Add these insertions to `data.sql`.
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Verify that changes were made.
- Commit the transaction.
- Verify that changes persist after commit.
- Take a screenshot of the results of actions.
- Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists.
- Take a screenshot of the results of actions.
- Inside a transaction:
   - Delete all animals born after Jan 1st, 2022.
   - Create a savepoint for the transaction.
   - Update all animals' weight to be their weight multiplied by -1.
   - Rollback to the savepoint
   - Update all animals' weights that are negative to be their weight multiplied by -1.
   - Commit transaction
- Take a screenshot of the results of actions.
- Add all these transactions to `queries.sql` file.
- Write queries to answer the following questions:
   - How many animals are there?
   - How many animals have never tried to escape?
   - What is the average weight of animals?
   - Who escapes the most, neutered or not neutered animals?
   - What is the minimum and maximum weight of each type of animal?
   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
   - Take a screenshot of the results of your queries.
- Add these queries to `queries.sql` file.
- Add screenshots with the results of your queries and with the results of play with transactions to Pull Request description.

 SQL_query-2 1
SQL_query-2 2
SQL_query-2 3
SQL_query-2 4
SQL_query-2 5
SQL_query-2 6 2
SQL_query-2 7
SQL_query-2 8
SQL_query-2 9
SQL_query-2 10
SQL_query-2 11
SQL_query-2 12
SQL_query-2 13
SQL_query-2 14

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,4 +5,14 @@ VALUES
...
\ No newline at end of file
+('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, weight, neutered, escape_attempts)
+VALUES
+('Charmander', '2020-02-08', -11, false, 0),
+('Plantmon', '2021-11-15', -5.7, true, 2),
+('Squirtle', '1993-04-02', -12.13, false, 3),
+('Angemon', '2005-01-12', -45, true, 1),
+('Boarmon', '2005-01-07', 20.4, true, 7),
```
Comment (Ismailco) [inline data.sql]: - [ ] The data_of_birth is wrong here, kindly make sure to correct these values the month in these two entries is `Jun` so the data should be `2005/06/12` and `2005/06/07`

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,74 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight >= 10.4 AND weight <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight >= 10.4 AND weight <= 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
```
Comment (Ismailco) [inline queries.sql:17]: - [ ] Before the `ROLLBACK` make sure to check that the changes were made, then  after check the changes were undone

e.g
```sql
SELECT species from animals;
   ROLLBACK;

 SELECT species from animals;
```

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,74 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
```
Comment (Ismailco) [inline queries.sql:33]: - [ ] Same here  Before the `COMMIT` make sure to check that the changes were made, then after check the changes were undone


```sql
SELECT species from animals; 
   COMMIT;
SELECT species from animals;
```

File: queries.sql [hunk 4]
```diff
@@ -7,4 +7,74 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+MAX(weight) as Max_Weight
+FROM animals
+GROUP BY species;
+
+SELECT species,
+AVG(escape_attempts) as Avg_Escape_Attempts
+FROM animals
+WHERE
+date_of_birth >= '1990-01-01' AND
+date_of_birth <= '2000-01-01'
```
Comment (Ismailco) [inline queries.sql]: - [ ] Here you should select the last day of the year to capture the whole year, kindly consider fixing this so the data is 👉🏼 `2000/12/31`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @astroboyReloaded, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Used GitHub Flow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (astroboyReloaded) [reply data.sql]: 👍

File: queries.sql (hunk 4 above)
Comment (astroboyReloaded) [reply queries.sql:79]: I appreciate your suggestion, @Ismailco. But the comparison I used here is 'greater than or equal to' (>=), which represents the exact same thing as what you suggest. I like it better this way.

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @astroboyReloaded    , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**